### DISCUSSION TITLE
Create a "backlog" for open issues

### DISCUSSION THREAD

**Discussion by arnaudroques** [plantuml/plantuml#1337] [General]
Right now, we have more than 450 open issues.
Even if the vast majority of those issues are old, we are reluctant to close them, because when you close an issue you have only two choices:

  * close as completed (*done, closed, fixed, resolved*)
  * close as not planned (*won't fix, can't repro, duplicate, stale*)
  
So we cannot close the vast majority of those issues, because they do not enter in any of those categories.

**Furthermore, those issues are very important for us because they are some raw input of users needs.**
So we propose to create a new github project [backlog](https://github.com/plantuml/backlog) which would be a simple container for all those issues.

The idea is to simply manually transfert the issue using `transfert issue` button to the new [backlog](https://github.com/plantuml/backlog) project.

So the vast majority of open issues would move to [backlog](https://github.com/plantuml/backlog). Here (in the [backlog](https://github.com/plantuml/backlog) project) we may use specific labels (`sequencediagram`, `classdiagram`...) which would help us to sort and to manage this long list of issues.

And when we will decide to really work on such a [backlog](https://github.com/plantuml/backlog/issues) issue, we would move back the issue in the [plantuml](https://github.com/plantuml/plantuml/issues) project.
This way, open issues in [plantuml](https://github.com/plantuml/plantuml/issues) project will really be active.

What do you people think about it?

**Comment by matthew16550** [comment]
Always a difficult problem.

As an alternative we could try making a [GitHub Project](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/changing-the-layout-of-a-view) to show a subset of the current issues.  I haven't used this new project system much but it looks quite helpful.

The instructions say we choose which issues to show in the project and it can be automated via labels or other criteria.  So perhaps a Kanban board with two columns:
* `maybe-soon`
* `in-progress`

Use the `maybe-soon` label for issues you think should be visible on the board but no-one is actively working on.

Then the "backlog" is all the issues that don't have one of those labels.  We wouldn't need to move them - just don't label them.

We might try making a single Kanban with issues from many plantuml repos to give you a good overall view.

Also, the [good-first-issue](https://github.com/topics/good-first-issue) label is a common idea and could also appear in the Kanbans left column.

**Reply by Kernald** [reply]
I know I'm too late, but this would have been a much better solution. Even closing the bugs as stale would have been better IMHO - the current solution makes looking for existing bugs much harder than it should be, and most importantly, is a fairly uncommon pattern here on GitHub. This is going to cause confusion and duplicates, without achieving much - the issues search tools are pretty powerful and could have avoided dealing with a long list of issues, especially in combination of labels, projects...

**Comment by arnaudroques** [comment]
The good news is that you are not too late.

I'm going to try [Github Project](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/changing-the-layout-of-a-view) to see if it could help. Do you have any example of actual project using it? It could gives us some inspiration.

We _could_ move back all "backlog" issues to "plantuml" and put a `backlog` label on them. The big question is:
**Should we close those issues?**

**Reply by Kernald** [reply]
Examples of workflows pretty similar to what you described but leveraging projects that come to my mind are:
 - [Aspect Build](https://github.com/orgs/aspect-build/projects/6)
 - [Home Assistant](https://github.com/home-assistant/core/projects?type=classic) - especially their [Dev](https://github.com/home-assistant/core/projects/4) project.

[MUI](https://github.com/mui/material-ui/projects/25) uses a roadmap approach, with a "Future" column as a backlog.

[Mozilla Mobile](https://github.com/orgs/mozilla-mobile/projects/68) uses a classic triage/backlog/WIP/Done separation, with per-feature set projects (fairly granular).

IMHO closing issues is an option, but a better approach to just blindly closing them would be using something like [Stale](https://github.com/marketplace/stale). It's configurable, but a common workflow is to:

 - Add a comment on the issue, mentioning that it's stale and will be closed if no action happens in the next x days
 - If someone comments on the issue, nothing more happens
 - If nobody comments after x days, the issue is closed.

This leaves a chance to do a bit of triage organically. Closed issues are still showing up fairly easily when searching anyway, and can be re-opened - so it's not that big a deal either.

**Comment by arnaudroques** [comment]
Github Project is indeed very powerfull (thanks @matthew16550  for pointing this out and to @Kernald  to show us examples).

It's so powerfull that we will begin to do something simple to better understand the tool :-)

So here what we have in mind:
We are going to create two projects:
- [Work](https://github.com/orgs/plantuml/projects/3)
- [Backlog](https://github.com/orgs/plantuml/projects/4)

Only members of the teams and active contributors will have write access.
The default access will be only "read".

When a new issue is created, it will *not* be automatically added to the "Projects".
As today, a label "`triage`" is added to the issue, and here can happen:
- The issue could be directly closed if it's not relevant
- The issue could be labelled "`discussion`" if it's not precise enough
- The issue could be labelled "`question`" if we have asked some question and are waiting for answers

At some point, the issue is precise enough so that coding can occur.
At that time, we put the label "`Under evaluation`" so that developers know that it can be worked on.
(BTW, we are not 100% satisfied with "**under evaluation**" naming. We are open for suggestions!)
So, here, the issue automatically appears in the ["Under evaluation" columns](https://github.com/orgs/plantuml/projects/3/views/1).

Then the issue will go to "WIP" and "Done" when it is actually coded.

Issues that are interesting but that we keep for latter will be labelled "`backlog`" and put into the Backlog column of [Backlog](https://github.com/orgs/plantuml/projects/4).
(Probably we will add other column here to sort the backlog).

Those issue will be closed (so that they do not pollute the [issue view by default](https://github.com/plantuml/plantuml/issues)), but we will use Github Action to explain that the issue is indeed closed but that the original poster can still make comments. 

**But this time, before doing anything, we would like to know what you guys are thinking about it.**


PS: And we will [remove this](https://github.com/plantuml/backlog/issues) and put everything [back here](https://github.com/orgs/plantuml/projects/4/views/1).

**Reply by Kernald** [reply]
This makes sense to me. In terms of labels, I'm not entirely sold on `question` which sounds a bit like the issue is an actual question from a user - maybe `needs more info` would be a better option. For `Under evaluation`, I'd suggest `ready to implement`.

Those labels are definitely a minor point though, the overall idea seems much better than having a distinct backlog project :-)

**Reply by The-Lum** [reply]
Hi all,

- Did I understand well?
>[![](https://www.planttext.com/api/plantuml/svg/XP8zSiCW48LxJh7ZtXT8aV5uqgHJat6JXf-rJ8n1WYMQJCPt3yABB2T5gaNlVMzPGBj4CbA-EE7mI401ekrF1CP6r6I35oTh4F0oq1SOJ3hQeSgY10bX4mRqQ3OW4uIcKRIoPv44mHCugT0rOzCuxK9YJ0jTi4UzOSOcdLCgkrRKD7tCcEPPthr7D0ak8I9OVmmL6vSHknW-oc7W4EAvMXryD-_D6uo0dz9bIMs7ZFe_GboRVtZjgZJPOPLQr3lu4ZpU3h3Yxc9P5FRwP-gzr6SN-WefTq-gjGv2BPVSZ5dwTQ6bkmYmtJxNPvchvcguZi8yL2r6hWnaDZ8RvRjCtfXS5vEORatdCx6cEAyuhpYloaYAEobkiqDloY_v0m00)](https://www.planttext.com/?text=XP8zSiCW48LxJh7ZtXT8aV5uqgHJat6JXf-rJ8n1WYMQJCPt3yABB2T5gaNlVMzPGBj4CbA-EE7mI401ekrF1CP6r6I35oTh4F0oq1SOJ3hQeSgY10bX4mRqQ3OW4uIcKRIoPv44mHCugT0rOzCuxK9YJ0jTi4UzOSOcdLCgkrRKD7tCcEPPthr7D0ak8I9OVmmL6vSHknW-oc7W4EAvMXryD-_D6uo0dz9bIMs7ZFe_GboRVtZjgZJPOPLQr3lu4ZpU3h3Yxc9P5FRwP-gzr6SN-WefTq-gjGv2BPVSZ5dwTQ6bkmYmtJxNPvchvcguZi8yL2r6hWnaDZ8RvRjCtfXS5vEORatdCx6cEAyuhpYloaYAEobkiqDloY_v0m00)

Regards.
